### PR TITLE
spm: peripherals: Add rest of peripherals

### PR DESCRIPTION
**Pull request by sigvartmh**
Add the lacking peripherals to SPM. Change out put of `TWIM` and `SPIM`
to `SPI` and `TWI` as they would configure the same address to
non-secure. Even if you are using it in slave mode `TWIS/SPIS`.

Signed-off-by: Sigvart Hovland <sigvart.hovland@nordicsemi.no>

### CODE REVIEW

File: include/spm.rst [hunk 1]
```diff
@@ -47,12 +47,14 @@ Peripherals configured as Non-Secure
...
-   * SPIM3
+   * SPI1-3
    * TIMER0-2
-   * TWIM2
-   * UARTE0, UARTE1
+   * TWI1-3
+   * UARTE1, UARTE3
    * WDT
 
+If you want to turn UARTE0, TWI0, EGU0 and SPI0 non-secure ensure that they are not used by the SPM / Zephyr instance which SPM is using to setup the SPU.
```
Comment (ru-fu) [inline include/spm.rst]:
```suggestion
If you want to configure UARTE0, TWI0, EGU0, or SPI0 as Non-Secure, you must ensure that they are not used by the library instance that the :ref:`secure_partition_manager` sample uses to set up the SPU.
```

Not sure if that suggestion is technically correct though. Are we referring to this library with "SPM" here? Or to the sample?

Comment (sigvartmh) [reply include/spm.rst]: The sample/ the firmware which is running the SPM.

If you change the security attribute to either of the peripherals while being in this sample you'll get a bus fault as the addressing has changed from what you compiled with.

Comment (ru-fu) [reply include/spm.rst]: Sorry - I meant the SPM library or the SPM sample. 
But I *think* that's what my suggestion says? ;-)

Comment (sigvartmh) [reply include/spm.rst]: I think the order is reversed so you must ensure that they are not used by the sample. 

If we think of the SPM sample as consumer of the SPM library. Then the person configuring it with the SPM sample must ensure that it does not touch any of the peripherals which is being set to non-secure. If not since it's compiled for secure it will use the wrong addressing for peripherals.

Comment (sigvartmh) [reply include/spm.rst]: I'll try to incorporate your suggestion in the reversed order.

File: subsys/spm/Kconfig [hunk 2]
```diff
@@ -148,16 +160,52 @@ config SPM_NRF_EGU2_NS
...
+	bool "SPI1 is Non-Secure"
+	default y
+
+config SPM_NRF_SPIM2_NS
+	bool "SPI2 is Non-Secure"
+	default y
+
 config SPM_NRF_SPIM3_NS
-	bool "SPIM3 is Non-Secure"
+	bool "SPI3 is Non-Secure"
```
Comment (oyvindronningstad) [inline subsys/spm/Kconfig]: Why this change?

Comment (sigvartmh) [reply subsys/spm/Kconfig]: because SPIM and SPIS have the same address and aren't separately configurable secure/non-secure at least that's how I read the product specification.

Comment (oyvindronningstad) [reply subsys/spm/Kconfig]: I see. We should be more explicit about that in the printout or the Kconfig text. Or maybe `depends on` or `select` when two configs are linked like that.

Comment (sigvartmh) [reply subsys/spm/Kconfig]: So you want multiple configs pr SPI/I2C and other M/S ports?

Comment (oyvindronningstad) [reply subsys/spm/Kconfig]: Wait, is there no `SPM_NRF_SPI3_NS`? If so, why rename only the prompt and not the config as well? Mismatched name/prompt is confusing.

Anyway, I meant that if there are `SPM_NRF_*` configs which must be set to the same because of HW restrictions, that should be reflected in the Kconfig text, or (even better) enforced via `depends on` or `select`

**Comment by martiles** [comment]
This PR would benefit many developers so they do not have to manually address this if they are going to use a peripheral that is not available in non-secure area.  Please revisit this so we can get it into NCS master 👍

**Comment by hakonfam** [comment]
@sigvartmh please fix CI

**Comment by sigvartmh** [comment]
@hakonfam CI is green here now 🗡️

**Comment by jtguggedal** [comment]
It seems that this patch caused GPIOTE to stop working in non-secure apps.
`NRF_GPIOTE1` is for some reason not defined when `NRF_TRUSTZONE_NONSECURE` is not defined

**Comment by hakonfam** [comment]
> It seems that this patch caused GPIOTE to stop working in non-secure apps.
> `NRF_GPIOTE1` is for some reason not defined when `NRF_TRUSTZONE_NONSECURE` is not defined
> Ref: NCSDK-5530

Looking into it
